lib/globus_client/endpoint.rb

```diff
@@ -56,6 +56,11 @@ def disallow_writes
       access_request(permissions: "r")
     end
 
+    def get_filenames
+      filepath = File.join(config.uploads_directory, path).concat(PATH_SEPARATOR)
+      ls_path(filepath, [])
```

Comment:
I wonder if you could call `#full_path` here and skip hand-creating the `filepath` variable?

Reply:
Oh! That does work! I think this may have to get changed again if we refactor, but going with it for now.